mcs/class/monodoc/Test/Monodoc.Ecma/EcmaUrlTests.cs

```diff
@@ -454,38 +470,293 @@ public void ExplicitIndexerImplementation ()
 			AssertUrlDesc (ast, "P:System.Web.SessionState.HttpSessionStateContainer$System.Web.SessionState.IHttpSessionState.Item(System.Int32)");
 		}
 
-		/*		[Test]
-		public void TreeParsabilityTest ()
+		[Test]
+		public void ToEcmaCref_Namespace ()
 		{
-			var rootTree = RootTree.LoadTree ("/home/jeremie/monodoc/");
-			Node result;
-			var generator = new CheckGenerator ();
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Namespace,
+				Namespace = "System.IO"
+			};
 
-			foreach (var leaf in GetLeaves (rootTree.RootNode).Where (IsEcmaNode))
-				AssertUrl (leaf.PublicUrl);
+			AssertEcmaString ("N:System.IO", actual);
 		}
 
-		IEnumerable<Node> GetLeaves (Node node)
+		[Test]
+		public void ToEcmaCref_SimpleType ()
 		{
-			if (node == null)
-				yield break;
+			var actual = new EcmaDesc {
+				DescKind = EcmaDesc.Kind.Type,
```

Comment:
`EcmaDesc.Kind` is presumably a nested type, which should [normally be avoided](https://msdn.microsoft.com/en-us/library/ms229027(v=vs.110).aspx)

> **AVOID** publicly exposed nested types. The only exception to this is if variables of the nested type need to be declared only in rare scenarios such as subclassing or other advanced customization scenarios.

Hopefully that type isn't public...